src/main/kotlin/nexters/admin/service/attendance/AttendanceDtos.kt

```diff
@@ -0,0 +1,66 @@
+package nexters.admin.service.attendance
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import nexters.admin.domain.attendance.Attendance
+import nexters.admin.domain.attendance.AttendanceStatus
+import nexters.admin.domain.generation_member.GenerationMember
+import nexters.admin.domain.session.Session
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+data class FindAttendanceProfileResponse(
+        @get:JsonProperty(value = "isGenerationMember")
+        val isGenerationMember: Boolean,
+        val attendanceData: AttendanceProfileResponse?
+) {
+    companion object {
+        fun of(): FindAttendanceProfileResponse {
+            return FindAttendanceProfileResponse(false, null)
+        }
+
+        fun of(generationMember: GenerationMember, sessionToAttendance: Map<Session, Attendance>): FindAttendanceProfileResponse {
+            return FindAttendanceProfileResponse(true, AttendanceProfileResponse.of(generationMember, sessionToAttendance))
+        }
+    }
+}
+
+data class AttendanceProfileResponse(
+        val score: Int,
+        @get:JsonProperty(value = "isCompletable")
+        val isCompletable: Boolean,
+        val attendances: List<AttendanceResponse>
+) {
+    companion object {
+        fun of(generationMember: GenerationMember, sessionToAttendance: Map<Session, Attendance>): AttendanceProfileResponse {
+            return AttendanceProfileResponse(
+                    generationMember.score ?: 100,
+                    generationMember.isCompletable,
+                    sessionToAttendance.map { (session, attendance) ->
+                        AttendanceResponse.of(session, attendance)
+                    }
```

Comment:
취향 차이일 거 같은데 `it`키워드로 줄여쓸 수 있기는 해요. 다만, `session`, `attendance`를 명확하게 드러낼 수 있다는 장점이 줄어들긴 할 것 같군요

```suggestion
                    sessionToAttendance.map { AttendanceResponse.of(it.key, it.value) }
```

Reply:
저도 둘 중에 고민하다가 `session`, `attendance`를 명확하게 드러내는쪽이 나을 것 같아서 저렇게 개발했습니다!
다만, 다시 생각해보니 앞서 사용한 변수명 자체가 `sessionToAttendance`이기 때문에 `it.key`, `it.value`해도 `session`과 `attendance`임이 명확하게 드러나는 것 같네요!
제안해주신 방향으로 수정하겠습니다!